### DISCUSSION TITLE
Great potential

### DISCUSSION THREAD

**Discussion by mcavaciocchi** [Budibase/budibase#2299] [General]
Hi, thanks for the iniciative to create something awesome.

Here some of my experience until now.

1. In a screen with relation one-to-many, i could only have the relational data using a the Data Provider with the table (**1**) and a repeated FK, using the RelationShip provider, the relational data came blank.
My data is
Invoice - 1 to Many - InvoiceItem - 1 to Many - SellingItem

And in the same screen, i would like to be able to add more InvoiceItem, but could not find a way to do it.

![image](https://user-images.githubusercontent.com/74980167/128584979-ee11aa51-e64a-45af-a554-4c5a970d7831.png)

2. In InvoiceItems I have a calculated column that is the quantity of items plus the price of the product. {{times Quantity SellingItem.0.Unit Price}} , I tried to get the total price of the Invoice using another calculated column  is not working.

{{sum (pluck InvoiceItems "Total")}}

3. The table component is not sorting Date properly. (it would be nice to format the date)

![image](https://user-images.githubusercontent.com/74980167/128585363-de009002-fd38-4c47-9a9c-7b4d9f63e71b.png)
 
4. If i "inactivate" a Customer, i don't want that client to be selectable for new Invoices. Tried with view and filter in new Data Provider, but can't do it.

5. I see in some discussion about import csv with relation, it will spare lot of time, but till them, i create a powershell script to do it directly in CouchDB.
![image](https://user-images.githubusercontent.com/74980167/128585546-76dd0add-51f3-458e-b315-39c73f5203f7.png)

Well, i can see a huge future for Budibase! thanks again

**Comment by joebudi** [comment]
Thanks for the feedback @mcavaciocchi . I am going to share this with the team and one of them will respond with further info.

**Comment by shogunpurple** [comment]
Hi @mcavaciocchi, first of all thank you very much for the detailed feedback. We really appreciate it!

For your questions:

### 1. Relational Issues
You need to render a dataprovider for every nested level of your relationship. If you are not using a table, you need to make use of the `Repeater` component to render data. You can find more information about the Repeater component here:

https://docs.budibase.com/design/components/repeater

Modelling my data in the same way as you have, I can represent the invoice, invoiceItems and sellingItems in nested dataproviders, repeaters and tables.

![Screenshot 2021-08-09 at 17 06 56](https://user-images.githubusercontent.com/11256663/128737915-de2efea4-dc79-4be2-abea-105576e32ef8.png)

To add more SellingItems, you need to add another form in your Invoice detail screen, and set up the save action as follows so that the SellingItem is attached to the correct invoice.

![Screenshot 2021-08-09 at 16 51 16](https://user-images.githubusercontent.com/11256663/128735514-973d861e-e7e7-4973-8fd4-776bf628dbc2.png)


### 2. Formula field
Formula fields are generally for the calculation of a single row, rather than that of a whole table. Formulas are calculated at runtime, so you cannot use the result of a formula field (`Total` in this case) in another formula field, as you have done with invoices. The easiest way for you to represent this is to set up a repeater in the UI, and to use the same binding you just did:

`{{sum (pluck InvoiceItems "Total")}}` 

However, I am actually noticing an issue too where the formula fields are not being pulled back to the frontend. I've created an issue for this now so we can work on it.
    
### Date sorting issue
You can follow along with that particular issue here: https://github.com/Budibase/budibase/issues/2108

### Deactivating a customer, and not showing them in the filter
Just a question about how you are doing this - do you have a boolean field on a customer, and you are disabling that? It would be good to see how you are trying to do this, because it should be possible to do with a filter/data provider.

### Importing CSV with relations
This is something we haven't got round to yet, mainly because of how we would represent relationships in the CSV, maybe just mappings of IDs would work there. Is there a particular user experience you would like to see for this feature? A particular format you would like to use for your relationships?

Great job on that script you have put together, that's a nice way to solve it until we have something more concrete around importing relationships.

Thanks again for taking the time.

**Reply by mcavaciocchi** [reply]
Thanks for the reply. I’m now understanding much better how everything works. Keep the amazing job!

**Comment by hendrikjap** [comment]
Hi @shogunpurple , can you share this particular app or make it into a template? Thanks a lot mate!

**Reply by mjashanks** [reply]
Hey @ely2727 as this is a very old discussion, could you share more about what you are looking for. Does this help: https://docs.budibase.com/guides/create-a-child-record-from-the-parent ?

**Reply by hendrikjap** [reply]
Nvm, got it sorted mate! Thanks
